### ISSUE TITLE
Socket deadline

### ISSUE THREAD

**Issue by papey** [bearstech/ascetic-rpc#3]
On ascetic-rpc, if a function takes more than a second to excute, an EOF error occured.

We can simply demonstrate this using ascetic-rpc unit test, here is a diff : 

```
diff --git a/server/server_test.go b/server/server_test.go
index 274ead2..b51b439 100644
--- a/server/server_test.go
+++ b/server/server_test.go
@@ -4,6 +4,7 @@ import (
        "errors"
        "fmt"
        "testing"
+       "time"

        "github.com/bearstech/ascetic-rpc/client"
        "github.com/bearstech/ascetic-rpc/message"
@@ -44,6 +45,9 @@ func hello(req *message.Request) (*message.Response, error) {
        if err != nil {
                panic(err)
        }
+
+       time.Sleep(10 * time.Second)
+
        world := message.World{
                Message: fmt.Sprintf("Hello %s♥️", hello.Name),
        } 

```
I think main cause is this line https://github.com/bearstech/ascetic-rpc/blob/master/server/server.go#L86, used to close socket conn after 1 second.

**Comment by papey** [comment]
Regarding changes on commits https://github.com/bearstech/ascetic-rpc/commit/6ba4deebfbe9f3191da10e97a4588445ed30785d and https://github.com/bearstech/ascetic-rpc/commit/564d588ccb0e19e1622b8697b200233824f5ad2c. We now have to ensure timeout works correctly, see unit tests added in https://github.com/bearstech/ascetic-rpc/commit/d629b035f3a4dae649de0055a5c5932548079ee4 for more informations.

**Comment by athoune** [comment]
Fixed by 71a587fc732fd887d13747988d499baeec192ae0

The action is still alive, we need to use a Context to kill it when timeouting
